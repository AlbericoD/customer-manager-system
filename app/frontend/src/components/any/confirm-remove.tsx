import React, { FunctionComponent, useState } from 'react';
import { Confirm, Button, Icon } from 'semantic-ui-react';

export interface IConfirmRemove {
  remove: () => void;
  name: string;
}
export const ConfirmRemove: FunctionComponent<IConfirmRemove> = ({ remove, name }) => {
  const [open, toggle] = useState<boolean>(false);
  return (
    <>
      <Button color='youtube' onClick={() => toggle(true)}>
        <Icon name={'remove user'} />
      </Button>
      <Confirm
        open={open}
        onCancel={() => toggle(false)}
        onConfirm={remove}
        content={`Delete Customer: ${name} ?`}
      />
    </>
  );
};
