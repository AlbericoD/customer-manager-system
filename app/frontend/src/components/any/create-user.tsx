import React, { FunctionComponent, useEffect } from 'react';
import { Form, Button, Message, Icon, Transition } from 'semantic-ui-react';
import { ThunkDispatch } from 'redux-thunk';
import { connect } from 'react-redux';
import { AppState, resetForm, IResponse } from '../../store';
import { CustomerService } from '../../services';
import * as Input from '../inputs';
import './create-user.css';

interface IStateToProps {
  response: IResponse;
}
interface DispachProps {
  send: () => void;
  reset: () => void;
}
type Props = DispachProps & IStateToProps;

const FormUser: FunctionComponent<Props> = ({ response, send, reset }): JSX.Element => {
  useEffect(() => {
    reset();
  }, [reset]);
  return (
    <>
      <Transition visible={response.success} animation='scale' duration={400}>
        <Message success header='done' content='customer successfully created' />
      </Transition>
      <Transition visible={response.errors} animation='scale' duration={550}>
        <Message error header='needs revision' content='customer already exists' />
      </Transition>
      <Transition visible={!response.load} animation='scale' duration={700}>
        <Form
          onSubmit={(e: React.FormEvent<HTMLFormElement>) => {
            e.preventDefault();
            e.stopPropagation();
            send();
          }}
          loading={response.load}
          success={response.success}
          error={response.errors}>
          <Input.Name />
          <Input.Doc />
          <Input.Phone />
          <Input.Cel />
          <Input.Email />
          <Button basic color='yellow' inverted type='reset' onClick={() => reset()}>
            <Icon name='checkmark' />
            Reset
          </Button>
          <Button color='green' inverted type='submit'>
            <Icon name='checkmark' />
            Create
          </Button>
        </Form>
      </Transition>
    </>
  );
};
const mapStateToProps = (state: AppState): IStateToProps => ({
  response: state.customerReducer.response
});

const mapDispatchToProps = (dispatch: ThunkDispatch<{}, {}, any>): DispachProps => {
  return {
    send: () => {
      dispatch(CustomerService.instance.createCustomer());
    },
    reset: () => {
      dispatch(resetForm());
    }
  };
};
export const CreateUser = connect(
  mapStateToProps,
  mapDispatchToProps
)(FormUser);
