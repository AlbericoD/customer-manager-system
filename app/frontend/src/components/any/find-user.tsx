import React, { FunctionComponent, useState } from 'react';
import { Search } from 'semantic-ui-react';
import { ThunkDispatch } from 'redux-thunk';
import { connect } from 'react-redux';
import { AppState, setSelectedCustomer, ICustomer } from '../../store';
import { CustomerService } from '../../services';
import _ from 'lodash';

interface IStateToProps {
  customer: ICustomer;
  customers: ICustomer[];
}
interface DispachProps {
  selected: (customer: ICustomer) => void;
  search: (value: string) => void;
}
type Props = DispachProps & IStateToProps;

export interface IUser {
  avatar: string;
  name: string;
  doc: string;
  phone: string;
  email: string;
  cel: string;
}

//semantic ui tem um erro com o search component e props com bind
interface ITransformToSemanticSearch {
  id: number;

  title: string;
  description: string;
  price: string;
}
const transform = (users: IUser[]): ITransformToSemanticSearch[] =>
  users.map(
    (user, index): ITransformToSemanticSearch => ({
      id: index,
      title: user.name,
      description: user.email,
      price: user.doc
    })
  );

const Find: FunctionComponent<Props> = ({ selected, customer, customers, search }): JSX.Element => {
  const [isLoading, toggleLoading] = useState<boolean>(false);
  const [value, setValue] = useState<string>('');
  const initialState = () => {
    toggleLoading(false);
    setValue('');
    selected({ avatar: '', cel: '', doc: '', email: '', name: '', phone: '' });
  };
  const handleResultSelect = (e: any, { result }: any) => {
    setValue(result.title);

    selected({
      avatar: '',
      cel: '',
      doc: result.price,
      email: result.description,
      name: result.title,
      phone: ''
    });
  };

  const handleSearchChange = (e: any, { value }: any) => {
    toggleLoading(true);
    setValue(value);
    setTimeout(() => {
      if (value.length < 1) {
        return initialState();
      }
      search(value);
      toggleLoading(false);
    }, 300);
  };
  return (
    <Search
      minCharacters={2}
      size='small'
      fluid
      loading={isLoading}
      onResultSelect={handleResultSelect}
      onSearchChange={_.debounce(handleSearchChange, 500, {
        leading: true
      })}
      results={transform(customers)}
      value={value}
    />
  );
};
const mapStateToProps = (state: AppState): IStateToProps => ({
  customer: state.customerReducer.selected,
  customers: state.customerReducer.resultFind
});

const mapDispatchToProps = (dispatch: ThunkDispatch<{}, {}, any>): DispachProps => {
  return {
    selected: (customer: ICustomer) => {
      dispatch(setSelectedCustomer(customer));
    },
    search: (value: string) => {
      dispatch(CustomerService.instance.searchCustomer(value));
    }
  };
};
export const FindUser = connect(
  mapStateToProps,
  mapDispatchToProps
)(Find);
