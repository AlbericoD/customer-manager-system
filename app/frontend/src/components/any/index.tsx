export * from './requirements';
export * from './menu-action';
export * from './find-user';
export * from './create-user';
export * from './confirm-remove';
