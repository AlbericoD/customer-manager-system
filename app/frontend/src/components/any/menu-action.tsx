import React, { FunctionComponent, ReactNode, useState } from 'react';
import { Button, Menu, Icon, Modal, Header } from 'semantic-ui-react';

export interface IMenuAction {
  name: string;
  element: string;
  icon: any;
  sizeModal: any;
  header: string;
  buttonOK?: string;
  buttonOKType?: string;
  buttonCancel: string;
  children: ReactNode;
}

export const MenuAction: FunctionComponent<IMenuAction> = ({
  name,
  icon,
  element,
  sizeModal,
  header,
  buttonOK,
  buttonOKType,
  buttonCancel,
  children
}): JSX.Element => {
  const [open, toggle] = useState<boolean>(false);

  return (
    <>
      <Modal
        trigger={
          <Menu.Item as={element}>
            <Icon name={icon} />
            {name}
          </Menu.Item>
        }
        open={open}
        onOpen={() => toggle(true)}
        onClose={() => toggle(false)}
        basic
        size={sizeModal}>
        <Header icon='user' content={header} />
        <Modal.Content>{children}</Modal.Content>
        <Modal.Actions>
          <Button basic color='red' inverted onClick={() => toggle(false)}>
            <Icon name='remove' /> {buttonCancel}
          </Button>
          {buttonOK ? (
            <Button color='green' inverted type={buttonOKType}>
              <Icon name='checkmark' /> {buttonOK}
            </Button>
          ) : null}
        </Modal.Actions>
      </Modal>
    </>
  );
};
