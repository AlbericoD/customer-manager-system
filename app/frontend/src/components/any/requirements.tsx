import React, { FunctionComponent, ReactNode } from 'react';
import { Segment, Header } from 'semantic-ui-react';
export interface IRequiments {
  title: string;
  children: ReactNode;
}
export const Requiments: FunctionComponent<IRequiments> = ({ title, children }): JSX.Element => (
  <>
    <Header as='h3'>{title}</Header>
    <Segment>{children}</Segment>
  </>
);
