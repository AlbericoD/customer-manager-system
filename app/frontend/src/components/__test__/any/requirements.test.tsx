import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import Adapter from 'enzyme-adapter-react-16';
import { Requiments, IRequiments } from '../../any';

Enzyme.configure({
  adapter: new Adapter()
});

describe('Component Requiments', () => {
  let props: IRequiments = {
    title: 'test',
    children: <></>
  };
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Requiments {...props} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('render snapshot', () => {
    const component = shallow(<Requiments {...props} />);
    expect(component).toMatchSnapshot();
  });
});
