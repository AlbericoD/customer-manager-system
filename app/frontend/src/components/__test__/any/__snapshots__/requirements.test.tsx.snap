// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Requiments render snapshot 1`] = `
<Fragment>
  <Header
    as="h3"
  >
    test
  </Header>
  <Segment />
</Fragment>
`;
